app/controllers/avo/search_controller.rb

```diff
@@ -118,9 +127,36 @@ def apply_search_metadata(models, avo_resource)
       end
     end
 
-    def belongs_to_field
+
+    private
+
+    def fetch_parent
+      if params[:via_reflection_id].present?
+        @parent = params[:via_reflection_class].safe_constantize.find params[:via_reflection_id]
+      end
+    end
+
+    def should_apply_has_many_scope?
+      @should_apply_has_many_scope ||= params[:via_association] == 'has_many' && @resource.search_query.present?
+    end
+    
+    def should_apply_attach_scope?
+      @should_apply_attach_scope ||= params[:via_association] == 'belongs_to' && field_have_attach_scope?
```

Comment:
```suggestion
      params[:via_association] == 'belongs_to' && field_have_attach_scope?
```

Reply:
Here i think we should maintain the var because we don't want the field to get fetched more than once

Reply:
ok. let's keep it